### PR TITLE
Feature/tabs

### LINKED ISSUE #379: Tabs component
* Zeplin - https://zpl.io/VD4z6LW

- [x] grayish line on design is not considered part of the component
- [x] Check tests in vuetify for possible inspiration.
- [x] Perform component checks (#378)
- [ ] Add presentation page

### VaTabs
- [x] grow: boolean. Fill space vertically
- [x] right: boolean. align items to right
- [x] left: boolean. align items to left
- [x] hide-slider: boolean
- [x] value: number
- [x] should have default slot

### VaTab
- [x] disabled: boolean
- [x] should have default slot

### References
* Vuetify: https://vuetifyjs.com/en/components/tabs
* Quasar: https://quasar-framework.org/components/tabs.html

image

### Referencies
Story points: 3

### PR DESCRIPTION
**Pull request by vlad-shusterman**
close #379

### CODE REVIEW

File: src/components/dashboard/Dashboard.vue [hunk 1]
```diff
@@ -4,22 +4,23 @@
     <dashboard-info-widgets></dashboard-info-widgets>
 
     <vuestic-widget class="no-padding no-v-padding">
-      <vuestic-tabs
-        :names="[$t('dashboard.dataVisualization'), $t('dashboard.usersAndMembers'), $t('dashboard.setupProfile'), $t('dashboard.features')]"
+      <va-tabs
+        v-model="value"
+        :titles="[$t('dashboard.dataVisualization'), $t('dashboard.usersAndMembers'), $t('dashboard.setupProfile'), $t('dashboard.features')]"
```
Comment (asvae) [inline src/components/dashboard/Dashboard.vue]: Do you need them in both places?

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTab.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+</template>
+
+<script>
+export default {
+  name: 'va-tab',
+  props: {
+    value: null,
+    disabled: {
+      type: Boolean,
+      default: false
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTab.vue]: Likely you don't need that.

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <div class="va-tabs">
+    <div class="va-tabs__bar">
+      <div
+        v-for="title in titles"
+        class="va-tabs__bar-item"
+        :class="{'active': checkForActive(title)}"
+        @click="setActive(title)"
+        v-bind:key="title"
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: huh?

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      set (valueProxy) {
+        this.$emit('input', valueProxy)
+      },
+      get () {
+        return this.value
+      }
+    }
+  },
+  methods: {
+    setActive (value) {
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: Just do the check where you need it. No need to split into method.

**Comment by asvae** [comment]
❗ No link to issue.

**Comment by asvae** [comment]
> grow: boolean. Fill space vertically
  right: boolean. align items to right
  left: boolean. align items to left
  color: string: should allow theme color or direct color
  hide-slider: boolean

❗❗ Either not shown in demo or not done.

**Comment by asvae** [comment]
❗ https://cdn.discordapp.com/attachments/459745328212475914/533306113634861066/unknown.png

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTab.vue [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<template>
+  <div
+    @click="click"
```
Comment (AlexanderBoriskin) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTab.vue]: Make method name more meaningful

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      />
+    </div>
+    <div class="demo-container__item">
+      <vuestic-checkbox v-model="hideSlider" label="hide-slider"/>
+    </div>
+  </div>
+</template>
+
+<script>
+import VaTabs from './VaTabs'
```
Comment (AlexanderBoriskin) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: remove useless imports (after including components in vuestic-plugin.js)

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="va-tabs">
+    <div
+      class="va-tabs__bar"
+      :class="{
```
Comment (AlexanderBoriskin) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: BEM: va-tabs__bar--align-right, va-tabs__bar--grow
Fix there and below

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+       'align-right': right,
+       'grow': grow
+      }"
+    >
+      <div
+        class="va-tabs__bar-content"
+        :class="{'grow': grow}"
+        :style="{'background-color': color}"
```
Comment (AlexanderBoriskin) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: Use one naming for properties in style objects: 'background-color' -> backgroundColor.
Fix there and below.

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        <div
+          class="va-tabs__bar-content-items"
+          :class="{'grow': grow}"
+        >
+          <slot/>
+        </div>
+        <div
+          v-if="!hideSlider"
+          class="va-tabs__bar-content-slider"
+          :style="{
```
Comment (AlexanderBoriskin) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: Use computed property to simplify difficult expressions

**Comment by asvae** [comment]
Just opened the page...
image

**Comment by asvae** [comment]
Removing items keeps cursor
image

**Comment by asvae** [comment]
Try different text sizes
image

File: package-lock.json [hunk 10]
```diff
@@ -26,14 +26,14 @@
...
         "babylon": "7.0.0-beta.47",
-        "convert-source-map": "1.6.0",
-        "debug": "3.2.5",
-        "json5": "0.5.1",
-        "lodash": "4.17.11",
-        "micromatch": "2.3.11",
-        "resolve": "1.8.1",
-        "semver": "5.5.1",
-        "source-map": "0.5.7"
+        "convert-source-map": "^1.1.0",
```
Comment (asvae) [inline package-lock.json]: I'm pretty sure you don't have to update `package-lock.json` in scope of this task.

File: src/components/dashboard/Dashboard.vue [hunk 11]
```diff
@@ -1,29 +1,30 @@
...
+        </va-tab>
+        <va-tab>
+          {{$t('dashboard.setupProfile')}}
+        </va-tab>
+        <va-tab>
+          {{$t('dashboard.features')}}
+        </va-tab>
+      </va-tabs>
+      <data-visualisation-tab v-if="value === 'Data Visualization'"/>
+      <users-members-tab v-if="value === $t('dashboard.usersAndMembers')"/>
```
Comment (asvae) [inline src/components/dashboard/Dashboard.vue]: Imagine what would happen here if we hot change language.

File: src/components/dashboard/Dashboard.vue [hunk 12]
```diff
@@ -61,6 +61,11 @@ export default {
       })
     },
   },
+  data () {
+    return {
+      value: 'Data Visualization'
```
Comment (asvae) [inline src/components/dashboard/Dashboard.vue]: Normally you use keys `like-this` for navigation etc. Not translations.

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue [hunk 13]
```diff
@@ -0,0 +1,59 @@
+<template>
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: Should have scroll at least.

image

Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: image

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    VuesticCheckbox,
+  },
+  data () {
+    return {
+      count: 6,
+      value: 'item1item1item1item1item1',
+      options: ['default', 'align-right', 'grow'],
+      option: 'default',
+      color: 'White',
+      hideSlider: false
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: This should not influence bottom padding in any way. Purely visual

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue (hunk 13 above)
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: Selected newly added item
image

Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: Can't select 8th item
image

Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: On removal  selection moves to left. Should stay on same place.
image

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+        count++
+      }
+    })
+  }
+}
+</script>
+
+<style lang="scss">
+.va-tabs {
+  &__bar {
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.vue]: Is that additional wrapper any useful?

File: src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue (hunk 13 above)
Comment (asvae) [inline src/vuestic-theme/vuestic-components/vuestic-tabs/VaTabs.demo.vue]: Add some rescale condition for when slider goes from small to big tab.

**Pull request merged**